src/lib/bakeryjs/ComponentFactory.ts

```diff
@@ -77,13 +77,18 @@ export class ComponentFactory implements ComponentFactoryI {
 		componentsPath: string,
 		parentDir: string = ''
 	): void {
-		const files = fs.readdirSync(componentsPath);
+		const componentsPathSanit = componentsPath.endsWith('/')
+			? componentsPath
+			: `${componentsPath}/`;
+		const files = fs.readdirSync(componentsPathSanit);
```

Comment:
Another option would be to delegate this to module like `glob` or `globby` to recursively list files inside the directory.